stan/math/opencl/select_events.hpp

```diff
@@ -18,10 +18,6 @@ inline const std::vector<cl::Event>& select_events(const matrix_cl& m) {
   return m.events();
 }
 
-inline const std::vector<cl::Event>& select_events(matrix_cl* const& m) {
-  return m->events();
-}
-
```

Comment:
Do the tests pass for you when removing the above? I had this here for when we pass `this` in the sub_block etc. kernels that are in matrix_cl

Reply:
We'll see! They failed at compile time, right? I still can't actually run the tests but I can find compile errors on my laptop. 

Also, I think we should generally be using references and not pointers (except as required by opencl), so maybe we can fix that in sub_block etc

Reply:
Nevermind, `this` is always a pointer and there's no hacks I can find around it :'( I will add that back.

Reply:
> We'll see! They failed at compile time, right? I still can't actually run the tests but I can find compile errors on my laptop. Which test was it?

`./runTests.py -j15 test/unit/math/opencl/sub_block_test.cpp` should fail